accounts/views.py

```diff
@@ -20,7 +26,49 @@ def post(self, request, format='json'):
         serializer = UserSerializer(data=request.data)
         if serializer.is_valid():
             user = serializer.save()
+            # TODO temporary overloading is_active field as a mean to confirm the email.
+            # A better solution would be implement a profile model and add a is_email_confirmed
+            # field there.
+            user.is_active = False
+            user.save()
             if user:
+                url = reverse('activate_account',
+                              kwargs={
+                                  'uidb64': urlsafe_base64_encode(force_bytes(user.pk)).decode(),
+                                  'token': account_activation_token.make_token(user)
+                              },
+                              request=request)
+                message = f'''Hi {user.username},
+Please click on the link to confirm your registration,
+
+{url}'''
+                send_mail('Confirm your email', message, from_email='localhost', recipient_list=[user.email])
                 return Response(serializer.data, status.HTTP_201_CREATED)
 
         return Response(serializer.errors, status.HTTP_400_BAD_REQUEST)
+
+
+class ActivateAccount(APIView):
+    """API to activate account after signing-up.
+
+    get:
+    Confirm email address.
+    """
+
+    permission_classes = ()
+    authentication_classes = ()
```

Comment:
Are these permission_classes and authentication_classes place holders for future implementation? I don't see them used anywhere so I want to understand.

Reply:
This is because in the `MIT_backend/settings.py`, I set the defaults to:
```python
REST_FRAMEWORK = {
    'DEFAULT_PERMISSION_CLASSES': (
        'rest_framework.permissions.IsAuthenticated',
    ),
    'DEFAULT_AUTHENTICATION_CLASSES': (
        'rest_framework_jwt.authentication.JSONWebTokenAuthentication',
        'rest_framework.authentication.SessionAuthentication',
        'rest_framework.authentication.BasicAuthentication',
    ),
    'DEFAULT_RENDERER_CLASSES': (
        'rest_framework.renderers.JSONRenderer',
        'rest_framework.renderers.BrowsableAPIRenderer',
    ),
}
```

This means all the API resource classes are default to check:

1. Is this user authenticated?
1. Is this user has permission?

In the API resource you pointed, our app does not have to check the authenticity nor permission while registering or activating account. Thus, we need to override the settings to an empty tuple.

My thought is that - once our API grows, pretty much every one has to check the authenticity and permission, so it's cleaner to make it default. For the APIs don't need authenticity/permission, we'll tag them out explicitly, like this one.